README.md

```diff
@@ -1,239 +1,224 @@
++ Calculations could be in terms of “cents” (fallible as explained above)
++ In some cases, developers end up using Number instead, believing it to be mostly safe, but in practice causing bugs, even if tries take care of any issues involving rounding or non-exact conversions from decimals to binary floats
 
-If it works out reasonably to provide for it within the same proposal, it would also be nice to provide support for higher-precision applications of floating point numbers.
+#### Goals implied by the main use cases
 
-If Decimal is arbitrary-precision or supports greater precision than Number, it may also be used for applications which need very large floating point numbers, such as astronomical calculations, physics, or even certain games. In some sense, larger or arbitrary-precision binary floats (as supported by [QuickJS](https://bellard.org/quickjs/), or IEEE 754 128-bit/256-bit binary floats) may be more efficient, but Decimal may also be suitable.
+This use case implies the following goals:
 
-#### Sample code
++ Avoid unintentional rounding that causes user-visible errors
++ Basic mathematical functions such as `+`, `-`, `*`, and `/`
++ Sufficient precision for typical money and other human-readable quantities
++ Conversion to a string in a locale-sensitive manner
++ Sufficient ergonomics to enable correct usage
++ Be implementable with adequate performance/memory usage for applications
++ (Please file an issue to mention more requirements)
 
-For example, Fabrice Bellard wrote [this code](https://bellard.org/quickjs/pi_bigdecimal.js) to calculate digits of pi using BigDecimal.
+### Secondary use case: Data exchange
+
+In both frontend and backend settings, JavaScript is used to communicate with external systems, such as databases and foreign function interfaces to other programming languages. Many external systems already natively support decimal numbers. In such a setting, JavaScript is then the lower common denominator. With decimals in JavaScript, one has the confident that the numeric data one consumes and produces is handled exactly.
 
 #### Why use JavaScript for this case?
 
-(TODO)
+JavaScript is frequently used as a language to glue other systems together, whether in client, server or embedded applications. Its ease of programming and embedding, and ubiquity, lend itself to this sort of use case. Programmers often don’t have the option to choose another language. When decimals appear in these contexts, it adds more burden on the embedder to develop an application-specific way to handle things; such specificity makes things less composable.
 
-#### Goals implied by use case
+#### Goals implied by the use case
 
-This use case implies the following goals:
-- Basic mathematical functions such as `+`, `-`, `*`
-- Support of various numerical functions (e.g., trigonometric, log/exp, etc)
```

Comment:
I think these are still valid for the interaction use case.

Reply:
Good point. I've put them back in.